File: src/main/java/com/wanderers/hotelier_webservices/server/exception/HotelierServiceException.java
package com.wanderers.hotelier_webservices.server.exception;

import com.wanderers.hotelier_webservices.server.service.api.HotelierService;

/**
 * Custom Exception class which used to throw failures in {@link HotelierService}
 */
public class HotelierServiceException extends ServerException {

    private static final long serialVersionUID = 1L;

    /**
     * Class constructor
     *
     * @param message String value of Exception
     */
    public HotelierServiceException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: src/main/kotlin/com/wanderers/hotelier_webservices/server/exception/HotelierServiceException.kt
package com.wanderers.hotelier_webservices.server.exception

import com.wanderers.hotelier_webservices.server.service.api.HotelierService

/**
 * Custom Exception class which used to throw failures in [HotelierService]
 */
class HotelierServiceException
/**
 * Class constructor
 *
 * @param message String value of Exception
 */
    (message: String?, cause: Throwable?) :
    ServerException(message, cause) {
    companion object {
        private const val serialVersionUID = 1L
    }
}